pkg/state/continuity_check_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+2023 NVIDIA CORPORATION & AFFILIATES
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package state
+
+import (
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"gopkg.in/yaml.v3"
+
+	mellanoxv1alpha1 "github.com/Mellanox/network-operator/api/v1alpha1"
+	"github.com/Mellanox/network-operator/pkg/render"
+	"github.com/Mellanox/network-operator/pkg/utils"
+)
+
+func extractContainerNamesFromHelmChart(path string) ([]string, error) {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return nil, err
+	}
+
+	var parsedData map[string]interface{}
+	err = yaml.Unmarshal(data, &parsedData)
+	if err != nil {
+		return nil, err
+	}
+
+	containerNames := extractContainerNamesFromSubsection(parsedData)
+
+	return containerNames, nil
+}
+
+//nolint:gocognit
+func extractContainerNamesFromSubsection(data interface{}) []string {
+	var names []string
+
+	switch v := data.(type) {
+	case []interface{}:
+		for _, item := range v {
+			names = append(names, extractContainerNamesFromSubsection(item)...)
+		}
+	case map[string]interface{}:
+		for key, value := range v {
+			if key == "containerResources" {
```

Comment:
The 'containerResources' are commented out in the values.yaml. I believe it won't find any.

Reply:
Thank you! Fixed that by adding a function that uncomments containerResources sections